File: stack_python/stack.py
class Stack:

    def __init__(self):
        self.items = []

    def __len__(self):
        return len(self.items)

    def peek(self):
        return self.items[-1]

    def is_empty(self):
        return self.items == []

    def push(self, data):
        if data is None:
            raise TypeError('data cannot be None')
        self.items.append(data)

    def pop(self):
        if self.is_empty():
            raise Exception('stack is empty')
        return self.items.pop()

    def print_stack(self):
        print(self.items)


stack = Stack()
print(stack.is_empty())
stack.push('a')
stack.push('b')
stack.push('c')
stack.push(1)
stack.push(2)
print(stack.peek())
print(stack.__len__())
stack.print_stack()
stack.pop()
stack.print_stack()

File: stack_typescript/stack.js
var Stack = /** @class */ (function () {
  function Stack() {
    this.items = [];
  }
  Object.defineProperty(Stack.prototype, "length", {
    get: function () {
      return this.items.length;
    },
    enumerable: false,
    configurable: true,
  });
  Stack.prototype.peek = function () {
    return this.items[this.items.length - 1];
  };
  Stack.prototype.isEmpty = function () {
    return this.items.length === 0;
  };
  Stack.prototype.push = function (data) {
    if (data === null || data === undefined) {
      throw new Error("data cannot be null");
    }
    this.items.push(data);
  };
  Stack.prototype.pop = function () {
    if (this.isEmpty()) {
      throw new Error("stack is empty");
    }
    return this.items.pop();
  };
  Stack.prototype.printStack = function () {
    console.log(this.items);
  };
  return Stack;
})();
var stack = new Stack();
console.log(stack.isEmpty());
stack.push("a");
stack.push("b");
stack.push("c");
stack.push(1);
stack.push(2);
console.log(stack.peek());
console.log(stack.length);
stack.printStack();
stack.pop();
stack.printStack();
